ostd/src/task/scheduler/mod.rs

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#![allow(dead_code)]
+
+mod fifo_scheduler;
+
+pub use fifo_scheduler::FifoScheduler;
+use spin::Once;
+
+use super::{processor::switch_to_task, task::Task, TaskStatus};
+use crate::{arch::timer::register_callback, prelude::*};
+
+static SCHEDULER: Once<&'static dyn Scheduler<Task>> = Once::new();
+
+/// Injects a scheduler implementation into framework.
+///
+/// This function can only be called once and must be called during the initialization of kernel.
+pub fn inject_scheduler(scheduler: &'static dyn Scheduler) {
+    SCHEDULER.call_once(|| scheduler);
+
+    register_callback(|| {
+        SCHEDULER.get().unwrap().local_mut_rq_with(&mut |local_rq| {
+            if local_rq.update_current(UpdateFlags::Tick) {
+                local_rq.set_should_preempt(true);
```

Comment:
Again, it does not make sense to have this `set_should_preempt` method on local runqueues. Give this method to `Task`.